### ISSUE TITLE
Name clash in error message

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#3053]
```agda
open import Common.Product
open import Common.Equality

infixr 9 _∘_

_∘_ : ∀ {a b c}
        {A : Set a} {B : A → Set b} {C : {x : A} → B x → Set c} →
        (∀ {x} (y : B x) → C y) → (g : (x : A) → B x) →
        ((x : A) → C (g x))
f ∘ g = λ x → f (g x)

_×̇_ : ∀{A C : Set} {B : A → Set} {D : C → Set}
     (f : A → C) (g : ∀ a → B a → D (f a)) → Σ A B → Σ C D
(f ×̇ g) (x , y) = f x , g x y

mutual
  data Cxt : Set₁ where
    _▷_ : (Γ : Cxt) (U : C⦅ Γ ⦆ → Set) → Cxt

  C⦅_⦆ : Cxt → Set
  C⦅ Γ ▷ U ⦆ = Σ C⦅ Γ ⦆ U

mutual
  data Ty (Γ : Cxt) : Set where
    pi   : (A : Ty Γ) (B : Ty (Γ ▷ T⦅ A ⦆)) → Ty Γ

  T⦅_⦆ : ∀{Γ} (A : Ty Γ) → C⦅ Γ ⦆ → Set
  T⦅ pi A B ⦆ γ = (x : T⦅ A ⦆ γ) → T⦅ B ⦆ (γ , x)


mutual
  inst : ∀{Γ} (A : Ty Γ) Δ (f : C⦅ Δ ⦆ → C⦅ Γ ⦆) → Ty Δ
  inst (pi A B) Δ f = pi (inst A Δ f)
                         (inst B (Δ ▷ T⦅ inst A Δ f ⦆) (f ×̇ denT-inst A Δ f))

  denT-inst : ∀{Γ} (A : Ty Γ) Δ (f : C⦅ Δ ⦆ → C⦅ Γ ⦆) (δ : C⦅ Δ ⦆) → T⦅ inst A Δ f ⦆ δ → T⦅ A ⦆ (f δ)
  denT-inst A Δ f δ x = subst (λ z → z δ) (denT-inst-eq A Δ f) x

  denT-inst-eq : ∀{Γ} (A : Ty Γ) Δ (f : C⦅ Δ ⦆ → C⦅ Γ ⦆) → T⦅ inst A Δ f ⦆ ≡ T⦅ A ⦆ ∘ f
  denT-inst-eq {Γ} (pi A B) Δ f
    with T⦅ inst A Δ f ⦆
       | denT-inst-eq A Δ f
       | T⦅ inst B (Δ ▷ T⦅ inst A Δ f ⦆) (f ×̇ denT-inst A Δ f) ⦆
       | T⦅ B ⦆ ∘ (f ×̇ denT-inst A Δ f)
       | denT-inst-eq B (Δ ▷ T⦅ inst A Δ f ⦆) (f ×̇ denT-inst A Δ f)
  denT-inst-eq {Γ} (pi A B) Δ f | .(λ x → T⦅ A ⦆ (f x)) | refl | B' | .B' | refl = refl -- Error intended

-- The error message contains a name clash:
--
-- B' (x , x) != T⦅ B ⦆ (f x , x) of type Set
-- when checking that the expression refl has type
-- (λ γ → (x : T⦅ A ⦆ (f γ)) → B' (γ , x)) ≡
-- (λ x → (x₁ : T⦅ A ⦆ (f x)) → T⦅ B ⦆ (f x , x₁))
```
Agda-2.5.1.1 has no such name clash, it reports:
```
B' (x , x₁) != T⦅ B ⦆ (f x , x₁) of type Set
```
Possibly related: #2522 ?!

**Comment by nad** [comment]
> Possibly related: #2522 ?!

Bisection suggests that the answer is yes, so I'm closing this ticket.

**Comment by andreasabel** [comment]
The OP was apparently fixed in 2.6.1.
